test/extensions/tracers/datadog/event_scheduler_test.cc

```diff
@@ -0,0 +1,111 @@
+#include <chrono>
+
+#include "source/extensions/tracers/datadog/event_scheduler.h"
+
+#include "test/mocks/event/mocks.h"
+#include "test/mocks/thread_local/mocks.h"
+
+#include "datadog/json.hpp"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace Datadog {
+namespace {
+
+TEST(DatadogEventSchedulerTest, ScheduleRecurringEventCallsCreatesATimer) {
+  testing::NiceMock<ThreadLocal::MockInstance> thread_local_storage_;
+
+  EventScheduler scheduler{thread_local_storage_.dispatcher_};
+  testing::MockFunction<void()> callback;
+  // The interval is arbitrary in these tests; we just have to be able to
+  // compare it to what was passed to the mocks.
+  // The only requirement is that it be divisible by milliseconds, because
+  // that's what `Timer::enableTimer` accepts.
+  const auto interval = std::chrono::milliseconds(2000);
+
+  EXPECT_CALL(thread_local_storage_.dispatcher_, createTimer_(testing::_));
+
+  (void)scheduler.schedule_recurring_event(interval, callback.AsStdFunction());
```

Comment:
nit: in Envoy (for whatever reason) we do not use this syntax; we just drop the void. IMO it does convey useful info but that's not the official style.